### PR TITLE
bug fixes , minor improvements

### LINKED ISSUE #218: empty strings are not respected in textEdits
**Description:**
When supplying an empty string in an textEdit (for example in a completion), the text edit is ignored by lsp4intellij. It is therefore impossible to delete text via a textedit.

Empty strings is specifically prevented by lsp4intellij in https://github.com/ballerina-platform/lsp4intellij/blob/master/src/main/java/org/wso2/lsp4intellij/editor/EditorEventManager.java#L1132, but i cannot see the reason for it, especially as there is code later on to handle empty strings. In my testing i did not found any problems. Is it safe to remove the check?

### LINKED ISSUE #226: Can LanguageServers freeze the UI?
**Description:**
While developing our internal language server, i notice that sometimes the ui froze. This is releated to our language server doing to much work and this is a separate issue of course. But a language server should never block an UI. 

I've attached a typical threaddump:
[threadDump-20201119-162826.txt](https://github.com/ballerina-platform/lsp4intellij/files/5573247/threadDump-20201119-162826.txt)

Unfortunately i'm not extremly well versed in Java request modeling, but i suspect that `didChange` somehow is blocking even tough it's just a notification which does not receive an answer according to LSP spec. Any ideas where the blocking occurs?

### PR DESCRIPTION
**Pull request by Trias**
## Purpose
fixes #226, fixes #218

also a follow up to https://github.com/ballerina-platform/lsp4intellij/pull/214/commits/818a39c0b18982602af833e2d414a70c1934afb8 as the solution had its problems, but now i added more control over the completion step, without intellij being "smart" about it, which makes completions by the languageserver more predictable.

also fixes a number of threading issues.

also adds the ability to restart the language server from the ui.

Please tell if you want me to rebase the commits, currently commits are a bit mixed up.

### CODE REVIEW

File: src/main/java/org/wso2/lsp4intellij/client/languageserver/wrapper/LanguageServerWrapper.java [hunk 1]
```diff
@@ -114,8 +114,7 @@
 import static org.wso2.lsp4intellij.requests.Timeout.getTimeout;
 import static org.wso2.lsp4intellij.requests.Timeouts.INIT;
 import static org.wso2.lsp4intellij.requests.Timeouts.SHUTDOWN;
-import static org.wso2.lsp4intellij.utils.ApplicationUtils.computableReadAction;
-import static org.wso2.lsp4intellij.utils.ApplicationUtils.invokeLater;
+import static org.wso2.lsp4intellij.utils.ApplicationUtils.*;
```
Comment (nixel2007) [inline src/main/java/org/wso2/lsp4intellij/client/languageserver/wrapper/LanguageServerWrapper.java:117]: please, do not use star-imports

Comment (Trias) [reply src/main/java/org/wso2/lsp4intellij/client/languageserver/wrapper/LanguageServerWrapper.java:117]: @nixel2007 It was done automatically by IntelliJ after importing `ApplicationUtils.pool`. Why should you not use star imports?

Comment (nixel2007) [reply src/main/java/org/wso2/lsp4intellij/client/languageserver/wrapper/LanguageServerWrapper.java:117]: https://rules.sonarsource.com/java/RSPEC-2208

**Comment by Trias** [comment]
I'll resubmit this PR as separate PR's split by feature

**Pull request closed without merging**